lib/__init__.py

```diff
@@ -53,11 +53,16 @@ def module_init():
 
     SOURCEDIR = os.path.dirname(os.getcwd())
     BUILDDIR = args.builddir
+    LUATEST_DIR = SOURCEDIR + "/luatest"
     os.environ["SOURCEDIR"] = SOURCEDIR
     os.environ["BUILDDIR"] = BUILDDIR
     soext = sys.platform == 'darwin' and 'dylib' or 'so'
-    os.environ["LUA_PATH"] = SOURCEDIR+"/?.lua;"+SOURCEDIR+"/?/init.lua;;"
-    os.environ["LUA_CPATH"] = BUILDDIR+"/?."+soext+";;"
+    os.environ["LUA_PATH"] = (LUATEST_DIR + "/?.lua;"
+                              + SOURCEDIR + "/test/?.lua;"
+                              + SOURCEDIR + "/?.lua;"
+                              + SOURCEDIR + "/?/init.lua;;")
+    os.environ['LUATEST_BIN'] = LUATEST_DIR + "/bin/luatest"
+    os.environ["LUA_CPATH"] = BUILDDIR + "/?." + soext + ";;"
```

Comment:
To be honest, I completely missed the point of this commit. Please, explain which problems are solved here in the commit message.

I guess that one of relevant problems is where to find luatest executable. There are two cases:

1. luatest is installed into the system (rare, but possible; personally I use this variant).
2. luatest is installed using `tarantoolctl rocks install luatest` into `.rocks` directory within current directory (source, build, cwd?) or somewhere upward.

If we'll add `.rocks/bin` (or `../../<...>/../.rocks/bin`) into PATH when it is found, we'll support both cases. I implemented this approach [here](https://github.com/tarantool/conf/blob/e1da1194fd247405e909a76127e31e0866efc179/Makefile#L17-L33). I hope it'll be easier in Python comparing to GNU Make.

About LUA_PATH changes: I need a problem description to don't repeat your effort.

Reply:
If luatest is installed by `tarantoolctl rocks install luatest` (case 2 above), `require('luatest')` should work without any LUA_PATH changes. But maybe I miss some cases: say, `--vardir` usage. I propose to describe your layout, so we can spot a problem and discuss how to better resolve it.

Reply:
As we discussed, now the script is searching for the `.rocks` directory in source and build directories.

Reply:
It is not exactly what I wanted:

1. I think that it should be either build or source directory. Otherwise the behaviour will be less obvious.
2. It should traverse directories upward recursively till the root directory, just like tarantool does for rocks.
3. I would want to keep ability to use /usr/bin/luatest + /usr/share/tarantool/luatest.

Aside of this, as stated [here](https://github.com/tarantool/tarantool/pull/6181/files#r680340435), we'll need to bundle luatest code into tarantool's source tarball. The easiest way to do so is to add luatest as submodule to test-run.

After enabling the submodule there is no much point to search luatest somewhere else. So my points above look non actual.

If you don't mind, I'll propose a set of small related patches.

Reply:
Ouch, I see now, it is recursive. Sorry.